mllib/src/main/scala/org/apache/spark/ml/evaluation/BinaryClassificationEvaluator.scala

```diff
@@ -36,12 +36,18 @@ import org.apache.spark.sql.types.DoubleType
 @Since("1.2.0")
 @Experimental
 class BinaryClassificationEvaluator @Since("1.4.0") (@Since("1.4.0") override val uid: String)
-  extends Evaluator with HasRawPredictionCol with HasLabelCol with DefaultParamsWritable {
+  extends Evaluator with HasRawPredictionCol with HasLabelCol
+    with HasWeightCol with DefaultParamsWritable {
 
   @Since("1.2.0")
   def this() = this(Identifiable.randomUID("binEval"))
 
   /**
+   * Default number of bins to use for binary classification evaluation.
+   */
+  val defaultNumberOfBins = 1000
```

Comment:
Why 1000?

Reply:
I think this should be a default in BinaryClassificationMetrics then?

Reply:
I'm not sure this was resolved; can the metrics class just use this as a default?

Reply:
done, made it default in BinaryClassificationMetrics